cmd/eksctl-anywhere/cmd/options.go

```diff
@@ -110,31 +116,28 @@ func readAndValidateClusterSpec(clusterConfigPath string, cliVersion version.Inf
 }
 
 func newClusterSpec(options clusterOptions) (*cluster.Spec, error) {
-	var specOpts []cluster.SpecOpt
+	var opts []cluster.FileSpecBuilderOpt
 	if options.bundlesOverride != "" {
-		specOpts = append(specOpts, cluster.WithOverrideBundlesManifest(options.bundlesOverride))
-	}
-	if options.managementKubeconfig != "" {
-		managementCluster, err := cluster.LoadManagement(options.managementKubeconfig)
-		if err != nil {
-			return nil, fmt.Errorf("unable to get management cluster from kubeconfig: %v", err)
-		}
-		specOpts = append(specOpts, cluster.WithManagementCluster(managementCluster))
+		opts = append(opts, cluster.WithOverrideBundlesManifest(options.bundlesOverride))
 	}
 
-	clusterSpec, err := readAndValidateClusterSpec(options.fileName, version.Get(), specOpts...)
+	clusterSpec, err := readAndValidateClusterSpec(options.fileName, version.Get(), opts...)
 	if err != nil {
 		return nil, fmt.Errorf("unable to get cluster config from file: %v", err)
 	}
 
 	if clusterSpec.Cluster.IsManaged() && options.managementKubeconfig == "" {
 		options.managementKubeconfig = kubeconfig.FromEnvironment()
-		clusterSpec.ManagementCluster = &types.Cluster{
-			Name:               clusterSpec.Cluster.ManagedBy(),
-			KubeconfigFile:     options.managementKubeconfig,
-			ExistingManagement: true,
+	}
```

Comment:
what if it's a workload cluster spec and management kubeconfig is empty? shouldn't we throw an error?

Reply:
I didn't change the logic here (or I didn't intend to)
That might be a validation we do somewhere else or one we don't do at all?

Reply:
i don't think we do it at all